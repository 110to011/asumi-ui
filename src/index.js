/**
 * Created by Elly on 2016/5/27.
 */

import Menu from './component/menu';
import SubMenu from './component/menu/subMenu';
import MenuItem from './component/menu/menuItem';
import MenuItemGroup from './component/menu/menuItemGroup';
import Tag from './component/tag';
import Grid from './component/grid';
import Form from './component/form';
import Tabs from './component/tabs';
import Radio from './component/radio';
import Group from './component/group';
import Input from './component/input';
import Modal from './component/modal';
import Table from './component/table';
import Editor from './component/editor';
import Col from './component/table/col';
import Button from './component/button';
import Select from './component/select';
import Loading from './component/loading';
import Popover from './component/popover';
import Message from './component/message';
import Animate from './component/animate';
import Upload from './component/upload';
import Checkbox from './component/checkbox';
import Dropdown from './component/dropdown';
import Option from './component/select/option';
import Pagination from './component/pagination';
import FormItem from './component/form/formItem';
import TabPanel from './component/tabs/tabPanel';
import RadioGroup from './component/radio/radioGroup';
import CheckboxGroup from './component/checkbox/checkGroup';
import SimplePagination from './component/pagination/simplePagination';
import DateTime from './component/datetime';

const Tooltip = Popover;

Table.Col = Col;

Select.Option = Option;

Tabs.TabPanel = TabPanel;

Menu.SubMenu = SubMenu;
Menu.MenuItem = MenuItem;
Menu.MenuItemGroup = MenuItemGroup;

Radio.Group = RadioGroup;
Checkbox.Group = CheckboxGroup;

Form.FormItem = FormItem;

Pagination.Simple = SimplePagination;

export {
    Col,
    Tag,
    Tabs,
    Menu,
    Form,
    Grid,
    Table,
    Modal,
    Input,
    Group,
    Radio,
    Select,
    Button,
    Upload,
    Option,
    Editor,
    SubMenu,
    Animate,
    Loading,
    Popover,
    Message,
    Tooltip,
    FormItem,
    MenuItem,
    TabPanel,
    Dropdown,
    Checkbox,
    DateTime,
    Pagination,
    RadioGroup,
    CheckboxGroup,
    MenuItemGroup,
    SimplePagination,
};