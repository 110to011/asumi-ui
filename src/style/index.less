@charset "utf-8";

@import "variables";
@import "normalize";
@import "fontawesome/less/font-awesome";

body {
  font-family: '微软雅黑', "Microsoft YaHei", Arial, Microsoft YaHei, Helvetica, sans-serif;
  margin: 0;
  color: @normalText;
  font-size: @fontSize;
  line-height: 1.42857;
}

a {
  color: @primary;
  &:hover, &:active, &:visited, &:focus {
    color: @d_primary;
  }
}

.el-shadow {
  box-shadow: 0 6px 12px @shadow;
  -webkit-box-shadow: 0 6px 12px @shadow;
}

.overflow {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@import "nav";
@import "tabs";
@import "drop";
@import "caret";
@import "modal";
@import "input";
@import "table";
@import "button";
@import "stretch";
@import "popover";
@import "checkbox";
@import "pagination";
