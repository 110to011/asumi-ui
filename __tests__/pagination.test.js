import React from 'react';
import {mount, shallow} from 'enzyme';
import {Pagination} from '../src';

describe('Pagination', () => {
    it('create component', () => {
        const component = shallow(<Pagination/>);
        expect(component).toMatchSnapshot();
    });
    it('define current & dataSize', () => {
        const component = mount(<Pagination current={2} dataSize={50}/>);
        expect(component).toMatchSnapshot();
    });
    it('define sizePerPage', () => {
        const component = mount(<Pagination sizePerPage={5} current={2} dataSize={50}/>);
        expect(component).toMatchSnapshot();
    });
    it('define paginationSize', () => {
        const component = mount(<Pagination paginationSize={5} current={2} dataSize={50}/>);
        expect(component).toMatchSnapshot();
    });
    it('define showTotalPages', () => {
        const component = mount(<Pagination showTotalPages={false}/>);
        expect(component).toMatchSnapshot();
    });
    it('define hideEndLabel & hideStartLabel', () => {
        const component = mount(<Pagination hideEndLabel hideStartLabel/>);
        expect(component).toMatchSnapshot();
    });
    it('define startLabel prevLabel nextLabel endLabel', () => {
        const component = mount(<Pagination startLabel="1" prevLabel="2" nextLabel="3" endLabel="4"/>);
        expect(component).toMatchSnapshot();
    });
});
