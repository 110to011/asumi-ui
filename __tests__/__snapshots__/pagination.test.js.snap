// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination create component 1`] = `
<div>
  <ul
    className="el-pagination"
  >
    <PageButton
      disabled={true}
      hidden={false}
      key="start"
      label="首页"
      onClick={[Function]}
      pgBtn={true}
    />
    <PageButton
      disabled={true}
      key="prev"
      label={
        <span>
          <span
            className="el-caret el-left"
          />
          上一页
        </span>
      }
      onClick={[Function]}
      pgBtn={true}
    />
    <PageButton
      disabled={true}
      key="next"
      label={
        <span>
          下一页
          <span
            className="el-caret el-right"
          />
        </span>
      }
      onClick={[Function]}
      pgBtn={true}
    />
    <PageButton
      disabled={false}
      hidden={false}
      key="end"
      label="尾页"
      onClick={[Function]}
      pgBtn={true}
    />
  </ul>
  <span
    className="el-totalPages"
  >
    共 
    0
     页
  </span>
</div>
`;

exports[`Pagination define current & dataSize 1`] = `
<Pagination
  current={2}
  dataSize={50}
  endLabel="尾页"
  hideEndLabel={false}
  hideStartLabel={false}
  nextLabel={
    <span>
      下一页
      <span
        className="el-caret el-right"
      />
    </span>
  }
  onPageChange={[Function]}
  paginationSize={6}
  prevLabel={
    <span>
      <span
        className="el-caret el-left"
      />
      上一页
    </span>
  }
  showTotalPages={true}
  sizePerPage={10}
  startLabel="首页"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={false}
        hidden={false}
        key="start"
        label="首页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              首页
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="prev"
        label={
          <span>
            <span
              className="el-caret el-left"
            />
            上一页
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                <span
                  className="el-caret el-left"
                />
                上一页
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="1"
        label={1}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={true}
        key="2"
        label={2}
        onClick={[Function]}
      >
        <li
          className="el-active"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="3"
        label={3}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              3
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="4"
        label={4}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              4
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="5"
        label={5}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              5
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="next"
        label={
          <span>
            下一页
            <span
              className="el-caret el-right"
            />
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                下一页
                <span
                  className="el-caret el-right"
                />
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={false}
        key="end"
        label="尾页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              尾页
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
    <span
      className="el-totalPages"
    >
      共 
      5
       页
    </span>
  </div>
</Pagination>
`;

exports[`Pagination define hideEndLabel & hideStartLabel 1`] = `
<Pagination
  current={1}
  dataSize={0}
  endLabel="尾页"
  hideEndLabel={true}
  hideStartLabel={true}
  nextLabel={
    <span>
      下一页
      <span
        className="el-caret el-right"
      />
    </span>
  }
  onPageChange={[Function]}
  paginationSize={6}
  prevLabel={
    <span>
      <span
        className="el-caret el-left"
      />
      上一页
    </span>
  }
  showTotalPages={true}
  sizePerPage={10}
  startLabel="首页"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={true}
        hidden={true}
        key="start"
        label="首页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-hidden el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              首页
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="prev"
        label={
          <span>
            <span
              className="el-caret el-left"
            />
            上一页
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                <span
                  className="el-caret el-left"
                />
                上一页
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="next"
        label={
          <span>
            下一页
            <span
              className="el-caret el-right"
            />
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                下一页
                <span
                  className="el-caret el-right"
                />
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={true}
        key="end"
        label="尾页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-hidden"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              尾页
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
    <span
      className="el-totalPages"
    >
      共 
      0
       页
    </span>
  </div>
</Pagination>
`;

exports[`Pagination define paginationSize 1`] = `
<Pagination
  current={2}
  dataSize={50}
  endLabel="尾页"
  hideEndLabel={false}
  hideStartLabel={false}
  nextLabel={
    <span>
      下一页
      <span
        className="el-caret el-right"
      />
    </span>
  }
  onPageChange={[Function]}
  paginationSize={5}
  prevLabel={
    <span>
      <span
        className="el-caret el-left"
      />
      上一页
    </span>
  }
  showTotalPages={true}
  sizePerPage={10}
  startLabel="首页"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={false}
        hidden={false}
        key="start"
        label="首页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              首页
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="prev"
        label={
          <span>
            <span
              className="el-caret el-left"
            />
            上一页
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                <span
                  className="el-caret el-left"
                />
                上一页
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="1"
        label={1}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={true}
        key="2"
        label={2}
        onClick={[Function]}
      >
        <li
          className="el-active"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="3"
        label={3}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              3
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="4"
        label={4}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              4
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="5"
        label={5}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              5
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="next"
        label={
          <span>
            下一页
            <span
              className="el-caret el-right"
            />
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                下一页
                <span
                  className="el-caret el-right"
                />
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={false}
        key="end"
        label="尾页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              尾页
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
    <span
      className="el-totalPages"
    >
      共 
      5
       页
    </span>
  </div>
</Pagination>
`;

exports[`Pagination define showTotalPages 1`] = `
<Pagination
  current={1}
  dataSize={0}
  endLabel="尾页"
  hideEndLabel={false}
  hideStartLabel={false}
  nextLabel={
    <span>
      下一页
      <span
        className="el-caret el-right"
      />
    </span>
  }
  onPageChange={[Function]}
  paginationSize={6}
  prevLabel={
    <span>
      <span
        className="el-caret el-left"
      />
      上一页
    </span>
  }
  showTotalPages={false}
  sizePerPage={10}
  startLabel="首页"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={true}
        hidden={false}
        key="start"
        label="首页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              首页
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="prev"
        label={
          <span>
            <span
              className="el-caret el-left"
            />
            上一页
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                <span
                  className="el-caret el-left"
                />
                上一页
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="next"
        label={
          <span>
            下一页
            <span
              className="el-caret el-right"
            />
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                下一页
                <span
                  className="el-caret el-right"
                />
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={false}
        key="end"
        label="尾页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              尾页
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
  </div>
</Pagination>
`;

exports[`Pagination define sizePerPage 1`] = `
<Pagination
  current={2}
  dataSize={50}
  endLabel="尾页"
  hideEndLabel={false}
  hideStartLabel={false}
  nextLabel={
    <span>
      下一页
      <span
        className="el-caret el-right"
      />
    </span>
  }
  onPageChange={[Function]}
  paginationSize={6}
  prevLabel={
    <span>
      <span
        className="el-caret el-left"
      />
      上一页
    </span>
  }
  showTotalPages={true}
  sizePerPage={5}
  startLabel="首页"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={false}
        hidden={false}
        key="start"
        label="首页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              首页
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="prev"
        label={
          <span>
            <span
              className="el-caret el-left"
            />
            上一页
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                <span
                  className="el-caret el-left"
                />
                上一页
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="1"
        label={1}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={true}
        key="2"
        label={2}
        onClick={[Function]}
      >
        <li
          className="el-active"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="3"
        label={3}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              3
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="4"
        label={4}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              4
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="5"
        label={5}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              5
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        active={false}
        key="6"
        label={6}
        onClick={[Function]}
      >
        <li
          className=""
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              6
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        key="next"
        label={
          <span>
            下一页
            <span
              className="el-caret el-right"
            />
          </span>
        }
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              <span>
                下一页
                <span
                  className="el-caret el-right"
                />
              </span>
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={false}
        key="end"
        label="尾页"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              尾页
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
    <span
      className="el-totalPages"
    >
      共 
      10
       页
    </span>
  </div>
</Pagination>
`;

exports[`Pagination define startLabel prevLabel nextLabel endLabel 1`] = `
<Pagination
  current={1}
  dataSize={0}
  endLabel="4"
  hideEndLabel={false}
  hideStartLabel={false}
  nextLabel="3"
  onPageChange={[Function]}
  paginationSize={6}
  prevLabel="2"
  showTotalPages={true}
  sizePerPage={10}
  startLabel="1"
>
  <div>
    <ul
      className="el-pagination"
    >
      <PageButton
        disabled={true}
        hidden={false}
        key="start"
        label="1"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="prev"
        label="2"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={true}
        key="next"
        label="3"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn el-disabled"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              3
            </span>
          </a>
        </li>
      </PageButton>
      <PageButton
        disabled={false}
        hidden={false}
        key="end"
        label="4"
        onClick={[Function]}
        pgBtn={true}
      >
        <li
          className="el-pg-btn"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            <span>
              4
            </span>
          </a>
        </li>
      </PageButton>
    </ul>
    <span
      className="el-totalPages"
    >
      共 
      0
       页
    </span>
  </div>
</Pagination>
`;
